ReactAndroid/src/main/java/com/facebook/react/views/image/ReactImageView.java

```diff
@@ -219,9 +278,7 @@ public void maybeUpdateView() {
 
   // VisibleForTesting
   public void setControllerListener(ControllerListener controllerListener) {
-    mControllerListener = controllerListener;
```

Comment:
This is used in tests only. Just to handle this case you allocate extra `forwardingcontroller` instance in the constructor. Instead we should have two member refs: `mControllerForTesting` and `mControllerListener` then set it in this method and in `setShouldNotifyOnLoadEvents`  respectively and only in `maybeUpdateView` we can allocate `forwardingController` in case when both are non-null (which will only happen in our instrumentation tests)